tests/integration/operator/operator_test.go

```diff
@@ -244,6 +246,20 @@ spec:
 	})
 })
 
+func ensureIstioCRsDeleted() error {
+	// This is a workaround to delete the Istio CRs that are left in the cluster
+	// This will be improved by splitting the tests into different Nodes with their independent setups and cleanups
+	crsToDelete := []string{"istio", "istiorevision"}
+	for _, cr := range crsToDelete {
+		err := kubectl.Delete(controlPlaneNamespace, cr, istioName)
```

Comment:
These resources are not namespaced, so `controlPlaneNamespace` is misleading here. I'm surprised that it even works. It's also possible that it will break at some point.

Reply:
I'd replace this for loop and just delete each resource separately (by specifying the kind and name for each particular resource). 

The fact that IstioRevision has the same name as the Istio resource is a coincidence and may even change in the future. A robust cleanup implementation would actually have to delete the IstioRevision via the ownerReference, not the name. But, having said that, we don't even need to delete the IstioRevision, because the kubernetes garbage collector should delete it when you delete the owner (the Istio resource).

The problem is that this deletion won't complete if the operator is broken, as it won't remove the finalizer. But that's another story, which we can probably ignore right now.